torch/csrc/jit/script/compiler.cpp

```diff
@@ -477,6 +478,12 @@ at::optional<std::vector<Value*>> tryMatchSchema(
           value = graph.insertNode(graph.createUndefined())->output();
       }
 
+      //implicit conversion of tensors to scalars
+      if(convert_tensors_to_nums && arg.type->isSubtypeOf(NumberType::get())
```

Comment:
The num to tensor case seems to be more complicated. Do we want methods to be able to be called on primitives - i.g.,  1.unsqueeze(0) 

Maybe we could not allow conversion on tensor arguments which are specified with "self" ?

Reply:
I added an error when methods are called on numbers. Separate from that - i was looking through aten ops and the cases I could think of where you would want num and tensors to be interchangeable already accept both. I think it would be better to identify the ops that should be interchangeable and aren't currently and add it to the op instead of the compiler.

Reply:
The behavior of the python frontend and script should match. So if the python frontend does num to tensor conversions, then we need to do them too. If it doesn't, then we shouldn't. I am not sure of the behavior but it should be easy to check.